kr8s/_api.py

```diff
@@ -508,19 +508,29 @@ async def async_watch(
         allow_unknown_type: bool = True,
     ) -> AsyncGenerator[tuple[str, APIObject]]:
         """Watch a Kubernetes resource."""
-        async with self.async_get_kind(
-            kind,
-            namespace=namespace,
-            label_selector=label_selector,
-            field_selector=field_selector,
-            params={"resourceVersion": since} if since else None,
-            watch=True,
-            timeout=None,
-            allow_unknown_type=allow_unknown_type,
-        ) as (obj_cls, response):
-            async for line in response.aiter_lines():
-                event = json.loads(line)
-                yield event["type"], obj_cls(event["object"], api=self)
+        while True:
+            async with self.async_get_kind(
+                kind,
+                namespace=namespace,
+                label_selector=label_selector,
+                field_selector=field_selector,
+                params={"resourceVersion": since} if since else None,
+                watch=True,
+                timeout=None,
+                allow_unknown_type=allow_unknown_type,
+            ) as (obj_cls, response):
+                async for line in response.aiter_lines():
+                    event = json.loads(line)
+                    if (
+                        event["object"]["kind"] == "Status"
+                        and event["object"].get("code") == 410
+                    ):
+                        break
+                    obj = obj_cls(event["object"], api=self)
+                    if since is not None:
```

Comment:
I think we still want to do this if `since` is `None`? Otherwise it will exit for the default case.

Reply:
done :+1: